lib/platform-api/requests.js

```diff
@@ -3,9 +3,40 @@ const platformApi = require('./client')
 const { isValidResponse } = require('../utils')
 const { chunk } = require('lodash')
 
+/**
+ *
+ *  Given an array of barcodes, resolves a map relating barcoes to M2 customer
+ *  codes (if found).
+ */
+const m2CustomerCodesForBarcodes = async (barcodes, offset = 0, map = {}, batchSize = 200) => {
+  if (!barcodes || barcodes.length === 0) return {}
+  // Batch calls to the m2-customer-code-store in groups of 200 barcodes:
+
+  const barcodesBatch = barcodes.slice(offset, offset + batchSize)
+  const client = await platformApi.client()
+  const response = await client.get(`m2-customer-codes?barcodes=${barcodesBatch.join(',')}`)
+
+  // Convert response to a map relating barcode to CC:
+  const responseAsMap = response.status === 400
```

Comment:
Just curious - did you encounter queries to the m2-customer-codes endpoint that responded with a 400?

Reply:
i wrote a 400 response for when theres no barcodes or zero customer code matches